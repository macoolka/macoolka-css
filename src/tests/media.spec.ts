

import {
    ExtendRule, RuleModule,
    parse as _parse, parseProp as _parseProp, Rule as ParentRule, ruleModule as ParentRuleModule, extendRule
} from '..';
/**
 * Enum Properties's value using string or number,
 * The tag 'ruleEnum' in rule describe the property style
 */
export type EProp = {
    size?: 'small' | 'medium'
};
/**
 * Stand Properties's value using any type,
 * The tag 'rule' in rule describe the property style
 */
export type SProp = {
    disabled?: boolean
};
export type Rule = ExtendRule<ParentRule, SProp, EProp>
export const rule: Rule = {
    rule: {
        disabled: ({ value }) => ({
            opacity: value ? 0.38 : 1,
        }),
    },
    /**
     * descirbe enum property.
     * the type is
     * propertyName :{propertyvalue:({value,name,theme,source})=>cssnode}
     */
    ruleEnum: {
        size: {
            small: () => ({
                mkFontSize: 'h1',
                mkHoverTextColor: 'accent',
                selector: {
                    ':focus': {
                        width: 10,
                    },
                },

            }),
            medium: () => ({
                mkFontSize: 'h2',
                selector: {
                    ':focus': {
                        width: 10,
                    },
                },
            }),

        },
    },
};
export const input: RuleModule<Rule> = {
    rule,
    theme: ParentRuleModule.theme,
    next: ParentRuleModule
}
export const mediaInput = extendRule(input)
export const parse = _parse(mediaInput);
describe('should parse component rule', () => {
    it('1 should parse empty props', () => {
        expect(parse({
            mkMedia: [{ size: 'small' }]
        })).toMatchSnapshot()
    });
    it('2 should parse common props', () => {
        expect(parse({
            mkMedia: [{ size: 'medium' }, { size: 'small' }, { size: 'small', disabled: false }]
        })).toMatchSnapshot()
    });

});