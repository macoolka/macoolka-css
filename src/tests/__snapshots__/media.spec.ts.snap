// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse component rule 1 should parse empty props 1`] = `
"@media screen and (max-width: 50em) {
  font-size: 96px;
}
@media screen and (max-width: 50em) {
  :focus {
    width: 10px;
  }
}
@media screen and (max-width: 50em) {
  :hover:not(:disabled) {
    color: #f50057;
  }
}
"
`;

exports[`should parse component rule 2 should parse common props 1`] = `
"@media screen and (max-width: 93em) {
  font-size: 96px;
  opacity: 1;
}
@media screen and (max-width: 80em) {
  font-size: 96px;
}
@media screen and (max-width: 50em) {
  font-size: 60px;
}
@media screen and (max-width: 93em) {
  :focus {
    width: 10px;
  }
}
@media screen and (max-width: 93em) {
  :hover:not(:disabled) {
    color: #f50057;
  }
}
@media screen and (max-width: 80em) {
  :focus {
    width: 10px;
  }
}
@media screen and (max-width: 80em) {
  :hover:not(:disabled) {
    color: #f50057;
  }
}
@media screen and (max-width: 50em) {
  :focus {
    width: 10px;
  }
}
"
`;
